### PR TITLE
fix(imagekit): transformation query param generation

### LINKED ISSUE #600: Imagekit provider not properly transforming images when using Firebase Storage (needs proper query string parameter)
When using the `imagekit` provider in Nuxt 3 version of this module, the transformation query string is prepended with a `?` parameter to the transformation like so: `?tr=w-200,h-200`. According to an [imagekit blog post](https://imagekit.io/blog/optimize-and-resize-images-in-firebase-storage/) it seems that you must use `&` instead of `?` like so:
`&tr=w-200,h-200` . Thus, an example working URL is:
 `https://ik.imagekit.io/ikmedia/o/product-clothes-3.jpg?alt=media&token=df8b1026-8c70-459d-87b3-1caa9f1d6d44&tr=w-300,h-400`

Unfortunately, when using the Nuxt Image module with Imagekit, the proper `&` parameter is not prepended to the transformation URL. For example, I have the module configured like so:

`nuxt.config.ts`:

```js
    image: {
        imagekit: {
          baseURL: 'https://ik.imagekit.io/XXX/'
        },
       domains: ['firebasestorage.googleapis.com'],
       screens: {
        xs: 0,
        sm: 576,
        md: 768,
        lg: 992,
        xl: 1200,
        xxl: 1400
       },
```
`Post.vue`:
```js
              <NuxtImg
                provider="imagekit"
                :src="post.pathURL" // "users%2FIdAUJMiswJajH8bnxdLrFpMhfTt1%2Fposts%2Fasdfdf_TIUI?alt=media&token=2598350b-3f09-41a4-8154-3d2df81435f2"
                width="48"
                height="48"
              />
```
The above code causes the image to be like so: 
`https://ik.imagekit.io/XXX/users%2FIdAUJMiswJajH8bnxdLrFpMhfTt1%2Fposts%2Fasdfdf_TIUI?alt=media&token=2598350b-3f09-41a4-8154-3d2df81435f2?tr=w-200,h-200` and although it works, it is not properly resized to 200x200. If you replace the `?` with the `&` in the URL transformation, it will be properly resized to 200x200.

Does anyone know how to fix this problem?

### PR DESCRIPTION
**Pull request by abhinavr4**
Fixes  #600

### CODE REVIEW

File: src/runtime/providers/imagekit.ts [hunk 1]
```diff
@@ -171,6 +171,14 @@ export const getImage: ProviderGetImage = (src, { modifiers = {}, baseURL = '/'
...
+  const url = new URL(joinURL(baseURL, src))
+
+  if (url.search) {
+    return {
+      url: joinURL(baseURL, src + (operations ? `&tr=${operations}` : ''))
+    }
+  }
+
   return {
     url: joinURL(baseURL, src + (operations ? `?tr=${operations}` : ''))
```
Comment (pi0) [inline src/runtime/providers/imagekit.ts:183]: I think here we can use `ufo.withQuery` to append params to the possible existing ones.

Comment (abhinavr4) [reply src/runtime/providers/imagekit.ts]: I have made the requested change.

**Comment by abhinavr4** [comment]
@pi0 Is there anything preventing this PR from being merged?

**Comment by dosstx** [comment]
Hi guys, any chance we can get this PR merged? Thank you!

**Review by pi0 — APPROVED** [review]
Thanks ❤️

**Pull request merged**